wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java

```diff
@@ -421,16 +421,29 @@ public Page createPage(Wiki wiki, String parentPageName, Page page) throws WikiE
       permissions = this.getWikiDefaultPermissions(wiki.getType(), wiki.getOwner());
       page.setPermissions(permissions);
     }
-    //init permission for page's creator
+    // init permission for page's creator
     String currentUser = org.exoplatform.wiki.utils.Utils.getCurrentUser();
-    if(!isExistedPermissionEntry(currentUser, permissions, page)) {
-      PermissionEntry permissionEntry = new PermissionEntry(currentUser, "", IDType.USER, new Permission[] {
-                                                                  new Permission(PermissionType.VIEWPAGE, true),
-                                                                  new Permission(PermissionType.EDITPAGE, true),
-                                                                  new Permission(PermissionType.ADMINPAGE, true) 
-                                                                  });
-      permissions.add(permissionEntry);
-      page.setPermissions(permissions);
+    PermissionEntry currentUserPermEnty = null;
+    for (PermissionEntry permEntry : page.getPermissions()) {
+      // get user permission entry
+      if (permEntry.getId().equals(currentUser)) {
+        currentUserPermEnty = permEntry;
+        if (!hasAdminPagePermission(permEntry)) {
+          // if user has no ADMINPAGE permission, add and update user PermissionEntry
+          List<Permission> userPermissions = Arrays.asList(permEntry.getPermissions());
+          userPermissions.add(new Permission(PermissionType.ADMINPAGE, true));

```

Comment:
There could already be an entry "Permission(PermissionType.ADMINPAGE, false)" in the list at this point. We should check that also IMO and update it if it is the case, to avoid any issue.
In fact you should probably directly loop over the permissions (so remove the method hasAdminPagePermission), and once you find the PermissionType.ADMINPAGE permission type, if it is false, you change it at true, if it is already true you break the loop, and if at the end of the loop the PermissionType.ADMINPAGE permission type has not been found, you add it.